Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertFloor.cs

```diff
@@ -196,38 +196,68 @@ private List<ICurve> GetProfiles(DB.CeilingAndFloor floor)
     }
 
     // in order to create a revit floor, we need to pass it a flat profile so change all the z values
-    private void ChangeZValuesForCurves(ICurve curve, double z)
+    private ICurve GetFlattenedCurve(ICurve curve, double z)
     {
+      // kind of a hack. Editing our speckle objects is so much easier than editing the revit objects
+      // so scale this z value from the model units to the incoming commit units and then those values will
+      // get scaled back to the model units when this entire outline gets scaled to native
+
       switch (curve)
       {
         case OG.Line line:
-          // kind of a hack. Editing our speckle objects is so much easier than editing the revit objects
-          // so scale this z value from the model units to the incoming units and then they will get scaled
-          // back to the model units when this entire outline gets scaled to native
-          line.start.z = z * Speckle.Core.Kits.Units.GetConversionFactor(ModelUnits, line.start.units);
-          line.end.z = z * Speckle.Core.Kits.Units.GetConversionFactor(ModelUnits, line.end.units);
-          return;
+          return new OG.Line(
+            new OG.Point(
+              line.start.x,
+              line.start.y,
+              z * Speckle.Core.Kits.Units.GetConversionFactor(ModelUnits, line.start.units),
+              line.start.units),
+            new OG.Point(
+              line.end.x,
+              line.end.y,
+              z * Speckle.Core.Kits.Units.GetConversionFactor(ModelUnits, line.end.units),
+              line.end.units),
+            line.units
+            );
 
         //case OG.Arc arc:
         //case OG.Circle circle:
         //case OG.Ellipse ellipse:
         //case OG.Spiral spiral:
 
         case OG.Curve nurbs:
-          for (int i = 2; i < nurbs.points.Count; i += 3)
-            nurbs.points[i] = z * Speckle.Core.Kits.Units.GetConversionFactor(ModelUnits, nurbs.units);
-          return;
+          var curvePoints = new List<double>();
+          for (var i = 0; i < nurbs.points.Count; i++)
+          {
+            if (i % 3 == 2)
+              curvePoints.Add(z * Speckle.Core.Kits.Units.GetConversionFactor(ModelUnits, nurbs.units));
+            else
+              curvePoints.Add(nurbs.points[i]);
+          }
+          var newCurve = OG.Curve.FromList(curvePoints);
+          newCurve.units = nurbs.units;
+          return newCurve;
```

Comment:
Same as the line. Computing the conversion factor just once (outside of the loop) will have some performance improvement when curves have a high number of points.